cineast-core/src/main/java/org/vitrivr/cineast/core/features/Superpixel.java

```diff
@@ -0,0 +1,256 @@
+   * BufferedImage
+   *
+   * @param segmentContainer original segmentContainer to apply superpixel algorithms
+   * @param algID            integer indicating which algorithm should be applied
+   * @return BufferedImage containing superpixels
+   */
+  public static BufferedImage applySuperpixelSC(SegmentContainer segmentContainer, int imgID,
+      int algID) {
+    BufferedImage image, mask;
+    switch (imgID) {
+      case 1:
+        image = segmentContainer.getAvgImg().getBufferedImage();
+        mask = segmentContainer.getAvgImg().getBufferedImage();
+        break;
+      case 2:
+        image = segmentContainer.getMedianImg().getBufferedImage();
+        mask = segmentContainer.getMedianImg().getBufferedImage();
+        break;
+      case 3:
+        image = segmentContainer.getMostRepresentativeFrame().getImage().getBufferedImage();
+        mask = segmentContainer.getMostRepresentativeFrame().getImage().getBufferedImage();
+        break;
+      default:
+        throw new IllegalStateException("Unexpected value: " + imgID);
+    }
+    return applySuperpixel(algID, image, mask);
+  }
+
+  /**
+   * This method applies the choice of the superpixel algorithm and computed superpixel algorithms
+   *
+   * @param algID integer indicating which algorithm should be applied
+   * @param image BufferedImage to apply superpixels to
+   * @param mask  BufferedImage containing original image to provide transparency mask
+   * @return BufferedImage containing superpixels
+   */
+  public static BufferedImage applySuperpixel(int algID, BufferedImage image, BufferedImage mask) {
+    image = ConvertBufferedImage.stripAlphaChannel(image);
+    ImageType<Planar<GrayF32>> imageType = ImageType.pl(3, GrayF32.class);
+
+    ImageSuperpixels alg;
+    switch (algID) {
+      case 1:
+        alg = FactoryImageSegmentation.fh04(new ConfigFh04(100, 30), imageType);
+        break;
+      case 2:
+        alg = FactoryImageSegmentation.meanShift(null, imageType);
+        break;
+      case 3:
+        alg = FactoryImageSegmentation.slic(new ConfigSlic(400), imageType);
+        break;
+      case 4:
+        alg = FactoryImageSegmentation.watershed(null, imageType);
+        break;
+      default:
+        throw new IllegalStateException("Unexpected value: " + algID);
+    }
+    ImageBase color = imageType.createImage(image.getWidth(), image.getHeight());
+    ConvertBufferedImage.convertFrom(image, color, true);
+    BufferedImage superpixel = Superpixel.performSegmentation(alg, color);
+
+    return Superpixel.applyTransparency(superpixel, mask);
+  }
+
+
+  /**
+   * Delete parts of superpixel images where sketch was transparent
+   *
+   * @param image superpixel sketch, transparent parts are coloured black
+   * @param mask  original sketch providing alpha (transparency) values
+   * @return cleaned BufferedImage of superpixels
+   */
+  public static BufferedImage applyTransparency(BufferedImage image, BufferedImage mask) {
```

Comment:
There is a proper way to deal with transparency and this is not it. If you supply an image with black strokes on a transparent background (which is not uncommon in a query) you destroy the entire information.